lib/matplotlib/axes/_axes.py

```
@@ -6054,25 +6054,30 @@ def _normalize_input(inp, ename='input'):
             """
             if (isinstance(x, np.ndarray) or
                     not iterable(cbook.safe_first_element(inp))):
-                # TODO: support masked arrays;
-                inp = np.asarray(inp)
+
+                inp = cbook.safe_masked_invalid(inp)
                 if inp.ndim == 2:
                     # 2-D input with columns as datasets; switch to rows
                     inp = inp.T
+
+                    if inp.shape[1] < inp.shape[0]:
```

Oh, wow, *that* I can see being a problem.
Maybe throw an error (or raise a warning) if x.shape[0] == 1?